src/include/metrics/query_trace_metric.h

```diff
@@ -111,10 +125,24 @@ class QueryTraceMetric : public AbstractMetric<QueryTraceMetricRawData> {
   friend class MetricsStore;
 
   void RecordQueryText(const execution::query_id_t query_id, const std::string &query_text, const uint64_t timestamp) {
-    GetRawData()->RecordQueryText(query_id, query_text, timestamp);
+    GetRawData()->RecordQueryText(query_id, "\"" + query_text + "\"", timestamp);
   }
-  void RecordQueryTrace(const execution::query_id_t query_id, const uint64_t timestamp) {
-    GetRawData()->RecordQueryTrace(query_id, timestamp);
+  void RecordQueryTrace(const execution::query_id_t query_id, const uint64_t timestamp,
+                        common::ManagedPointer<const std::vector<parser::ConstantValueExpression>> param,
+                        catalog::db_oid_t oid) {
+    std::ostringstream param_stream;
+    std::ostringstream type_stream;
+    for (const auto &val : (*param)) {
+      if (val.IsNull()) {
+        param_stream << "";
+      } else {
+        param_stream << val.ToString();
+      }
+      type_stream << type::TypeUtil::TypeIdToString(val.GetReturnValueType());
+      param_stream << ";";
+      type_stream << ";";
+    }
+    GetRawData()->RecordQueryTrace(query_id, timestamp, param_stream.str(), type_stream.str(), oid);
```

Comment:
I think we should move the type recording from the `QueryTrace` metric to the `QueryText` metric. The types should be fixed with the same query template and we only need to record them once and later load them once. Let me know whether this is doable.